src/main/java/me/zeroeightsix/kami/util/graphics/ShaderHelper.kt

```diff
@@ -44,9 +49,18 @@ class ShaderHelper(shaderIn: ResourceLocation, vararg frameBufferNames: String)
     // We are putting it here so it can find the listener above
     init {
         listener<ResolutionUpdateEvent> {
-            shader?.createBindFramebuffers(it.width, it.height)
+            shader?.createBindFramebuffers(it.width, it.height) // this will not run if on Intel GPU or unsupported Shaders
         }
 
         KamiEventBus.subscribe(this)
     }
+
+    companion object {
+        private var cachedIsIntegratedGraphics: Boolean? = null
```

Comment:
Why not just do this with `lateinit` and use `.isInitialized` to check

Reply:
It's in a companion object